### PR TITLE
Clam and virustotal

### PR DESCRIPTION
**Pull request by againer**
- make worker.rb more simple; added scanner and asset classes.
- add virustotal key to template.json

### CODE REVIEW

**Comment by againer** [comment]
DO NOT MERGE YET :)

**Comment by dfischer** [comment]
Check rubocop syntax

File: worker.rb [hunk 1]
```diff
@@ -2,87 +2,62 @@
...
+  # Scan each record available.
+  body['Records'].each do |record|
+    bucket = record['s3']['bucket']['name']
+    key = URI.decode(record['s3']['object']['key']).tr('+', ' ')
+    asset = Asset.new(bucket, key, TEMP_FILE, log)
+
+    # Set bucket and key for getting a bucket item.
+    asset.persist_local
+
+    next if Scanner.new.virus?(bucket, key, TEMP_FILE, log)
```
Comment (dfischer) [inline worker.rb:38]: might make more sense to pass those params to `new` instead of `virus?`

File: asset.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  end
+
+  # Delete the asset in s3.
+  def delete_remote
+    @s3.delete_object(bucket: @bucket, key: @key)
+  rescue StandardError => ex
+    @log.error("Caught #{ex.class} error calling delete_object on #{@key}.")
+  end
+
+  def delete_local
```
Comment (dfischer) [inline asset.rb:35]: not used?

File: scanner.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    elsif clam_result == 1
+      true
+    else
+      @log.debug("Error: ClamAV couldn't scan the file. Skipped #{key}")
+    end
+    false
+  end
+
+  def clam_scan(target)
+    @log.debug("scanning s3://#{bucket}/#{key} with clamav...")
```
Comment (dfischer) [inline scanner.rb:28]: `bucket` / `key` source?

**Comment by againer** [comment]
on hold

**Pull request closed without merging**